src/components/Checkout/CheckoutDropInPaymentWrapper.vue

```diff
@@ -168,7 +195,22 @@ export default {
 		},
 		validateGuestBasketAndCheckout() {
 			this.$emit('updating-totals', true);
-			this.validateGuestBasket(this.email, this.emailUpdates)
+			// Set the default checkout validation method
+			let validationMethod = this.validateGuestBasket;
+			const validationPayload = {
+				email: this.email,
+				emailUpdates: this.emailUpdates,
+			};
+			// If promo guest checkout is enabled, use the promo guest checkout validation method.
+			// This method validates the lender email for promo first before running the guest checkout method
+			// in checkout utils.
+			if (this.isGuestCheckout) {
```

Comment:
Should we be checking `this.isGuestCheckout` at the beginning of this method, and either wrapping the entire method content or returning early?

Reply:
I think this is supposed to be chekcout `promoGuestCheckoutEnabled` instead. If we're in this method, we already know we're in the guest checkout flow based on the method above.

This switch is between the mutation used to validate a promo with it's special conditions vs standard guest checkout.

Reply:
I'm not sure I understand what you're asking me here...  What do you mean by wrapping?
And why would I return early?  This is to set the method that is used in the lines below it so returning early would mean not calling the function below...

Reply:
I think @mcstover has more specific info that might be more appropriate. Basically what I was curious about was in this method for guest checkout, there is a check for `isGuestCheckout`. My assumption is that this method should only run if `isGuestCheckout` is `true`, so hence checking in the method before running any logic. But it sounds like maybe the check might just need to be a check on `promoGuestCheckoutEnabled`?

Reply:
So "isGuestCheckout" is the flag that enables the guest checkout user experience for the drop in payment wrapper.  The drop in payment wrapper is a component that gets used across kiva, not just for campaign pages.
promoGuestCheckoutEnabled is the flag that gets set in contentful that is **_specific to campaign pages_** so I don't want there to be a check for that in a generalized component.  I do set "isGuestCheckout" based on promoGuestCheckoutEnabled though in the parent component.

Reply:
<img width="514" alt="image" src="https://github.com/kiva/ui/assets/3431240/7188c1c2-6156-497e-99e4-fb16b6f81058">
See here ^

Reply:
Thanks for the info. It looks like the component has two props with the same value. Can those be consolidated?

Reply:
oh, thanks yeah that's not supposed to be there.

Reply:
oh wait, it is...
so sometimes is-guest-checkout will be the same check as promo-guest-checkout-enabled, but in other cases it might not be.  For example, if we are on kiva marketplace and not a corp campaign page, and a drop in payment wrapper is needed and it should follow the guest checkout, but not the promo-specific campaign copy.

Reply:
If I were to consolidate this, then the component wouldn't be able to tell the difference between a promo guest checkout workflow vs. a kiva marketplace guest checkout workflow.

Reply:
Thanks for the info. It seems like maybe `isGuestCheckout` is simply potentially a confusing name. If it only controls "guest checkout user experience for the drop in payment wrapper", the name might be more appropriate something along the lines of `isGuestCheckoutForDropIn`

Reply:
We'll have a look at this in some follow up work we're doing to see if we can make things more clear.